adaptivefiltering/segmentation.py

```diff
@@ -130,124 +174,189 @@ def __init__(self, dataset=None, segmentation=None):
+        :param azimuth:azimuth for the visualisation. Default = 315
+        :type azimuth: int
 
-        :param dataset:
-            The dataset from which the map should be displayed.
-        :type dataset: Dataset
+        :param angle_altitude: angle_altitude for the visualisation. Default = 45
+        :type angle_altitude: int
 
-        :return:
-            hexbin_segmentation:
-            The Segmentation of the area from the dataset
-        :type hexbin_segmentation: Segmentation
+        :param opacity: Sets the opacity of the layer, does not trigger recalculation of layers. Default = 0.6
+        :type opacity: float
 
         """
-        from adaptivefiltering.pdal import execute_pdal_pipeline, PDALInMemoryDataSet
-
-        # convert dataset to in memory pdal dataset
-        dataset = PDALInMemoryDataSet.convert(dataset)
-
-        # execute the reprojection and hexbin filter.
-        # this is nessesary for the map to function properly.
-        hexbin_pipeline = execute_pdal_pipeline(
-            dataset=dataset,
-            config=[
-                {"type": "filters.hexbin"},
-            ],
+        from adaptivefiltering.pdal import execute_pdal_pipeline
+
+        if self.dataset == None:
+            raise Exception("No dataset was given to calculate the hillshade or slope.")
+
+        if map_type != "Hillshade" and map_type != "Slope":
+            raise Exception("map_type can only be 'Hillshade' or 'Slope'.")
+
+        # set azimuth and angle_altitude to zero for _type =="Slope"
+        # This makes it easer to find preexisting slope overlays
+        if map_type == "Slope":
+            azimuth = 0
+            angle_altitude = 0
+
+        key_from_input = (
+            "_type:"
+            + map_type
+            + ",class:"
+            + str(classification)
+            + ",res:"
+            + str(resolution)
+            + ",az:"
+            + str(azimuth)
+            + ",ang:"
+            + str(angle_altitude)
         )
 
-        # get the coordinates from the metadata:
-        # this gives us lat, lon but for geojson we need lon, lat
-        coordinates = json.loads(hexbin_pipeline.metadata)["metadata"][
-            "filters.hexbin"
-        ]["boundary_json"]["coordinates"]
+        # if the dict is not empty, try to remove all layers present in the dict.
+        for layer in self.overlay_dict.values():
+            if layer.name == map_type:
+                try:
+                    self.map.remove_layer(layer)
+                except ipyleaflet.LayerException as e:
+                    continue
+
+        # if the desired hs is not already present, calculate it.
+        # if it is, it will simply be loaded at the end of the function.
+        if key_from_input not in self.overlay_dict.keys():
+            resolution = resolution
+
+            # calculate the hillshade or slope
+            if map_type == "Hillshade":
```

Comment:
Again this logic is duplicated from visualization and could be reused. Maybe the dispatch from https://github.com/ssciwr/adaptivefiltering/blob/hillshade_seg_map/adaptivefiltering/dataset.py#L114 needs to be refactored to be available from `visualization.py`.